packages/osd-i18n/src/core/i18n.ts

```diff
@@ -64,8 +74,16 @@ function getMessageById(id: string): string | undefined {
  * Normalizes locale to make it consistent with IntlMessageFormat locales
  * @param locale
  */
-function normalizeLocale(locale: string) {
-  return locale.toLowerCase();
+export function normalizeLocale(locale: string) {
+  const { lang, script, region } = localeParser.exec(locale)?.groups || {};
+  // If parsing failed or the language code was not extracted, return the locale
+  if (!lang) return locale;
```

Comment:
Is the right thing to accept the bad input or default in this case?

Reply:
This guy only does the normalization; (in)validation is handled by comparing it against those that have been registered.